### PR TITLE
Fido2 ext

### PR DESCRIPTION
**Pull request by conorpp**
Adds "solo" extension interface to FIDO2/CTAP2.

Similarly for U2F, custom commands can be added for the CTAP2 API.  Right now there are only two custom commands.

* `Get firmware version`
* `Get RNG bytes`

### CODE REVIEW

**Comment by conorpp** [comment]
Is CircleCI a concern?

**Comment by nickray** [comment]
no, circleci isn't setup right.

**Comment by nickray** [comment]
i removed circleci webhook.

File: targets/stm32l432/Makefile [hunk 1]
```diff
@@ -9,6 +9,25 @@ merge_hex=../../tools/solotool.py mergehex
 
 .PHONY: all all-hacker all-locked debugboot-app debugboot-boot boot-sig-checking boot-no-sig build-release-locked build-release build-release build-hacker build-debugboot clean clean2 flash flash_dfu flashboot detach cbor test
```
Comment (yparitcher) [inline targets/stm32l432/Makefile]: @conorpp Please add the new targets to the .PHONY line as the are not updateable files.
for example:
```
.PHONY: all all-hacker all-locked debugboot-app debugboot-boot boot-sig-checking boot-no-sig build-release-locked build-release build-release build-hacker build-debugboot clean clean2 flash flash_dfu flashboot detach cbor test firmware-hacker firmware-secure bootloader-nonverifying bootloader-verifying full-clean
```

Comment (conorpp) [reply targets/stm32l432/Makefile]: Will do, will add in future commit

**Pull request merged**